1 8 0 update build scan reports info (#3342)
JetBrains/kotlin-web-site: docs/topics/gradle/gradle-compilation-and-caches.md

>
{type="warning"}

Build reports for tracking compiler performance are available starting from Kotlin 1.7.0. Reports contain the durations 
of different compilation phases and reasons why compilation couldn't be incremental.

Use build reports to investigate performance issues when the compilation time is too long or when it differs for the same
project.

> Kotlin build reports help examine problems more efficiently than [Gradle build scans](https://scans.gradle.com/). 

Author: This and a paragraph below are taken from the blog post.

kotlin.build.report.label=some_label
```

### Limit of custom values

To collect build scans' statistics, Kotlin build reports use [Gradle's custom values](https://docs.gradle.com/enterprise/tutorials/extending-build-scans/). 
If you have a big project, a number of such custom values may be more than 1000 and these values consume a storage's space. 
Other Gradle plugins write custom values too. 
You can limit a maximum number of custom values in `gradle.properties` depending on your project's size:

```properties
kotlin.build.report.build_scan.custom_values_limit=1500
```

> If this amount is exceeded, you can see a following message in logs:

Author: ```suggestion
If this number is exceeded, you can see the following message in the logs:
```

```
Maximum number of custom values (1,500) exceeded
```

> 

Author: Or this variant:

To collect build scans' statistics, Kotlin build reports use [Gradle's custom values](https://docs.gradle.com/enterprise/tutorials/extending-build-scans/). 
If you have a big project, a number of such custom values may be quite big.

If number of custom values exceeds some value, you can see a following message in logs:

```
Maximum number of custom values (1,000) exceeded
```

To solve this, you can limit a maximum number of custom values in `gradle.properties` depending on your project's size:

```properties
kotlin.build.report.build_scan.custom_values_limit=1500
```

Keep in mind that these values consume a storage's space and other Gradle plugins write custom values too.

Author: Please choose one that you like more.
